packages/teams-js/src/private/filesExtensionsFor3PStorageproviders.ts

```diff
@@ -0,0 +1,194 @@
+      return;
+    }
+    getFilesDragAndDropViaCallback(dragAndDropInput, callback);
+  }
+
+  function getFilesDragAndDropViaCallback(
+    dragAndDropInput: string,
+    callback: (attachments: FilesFor3PApps[], error?: SdkError) => void,
+  ): void {
+    const files: FilesFor3PApps[] = [];
+    let helper: filesExtensionsFor3PStorageproviders.AttachmentListHelper = {
+      fileType: '',
+      assembleAttachment: [],
+    };
+
+    function handleGetDragAndDropFilesCallbackRequest(
+      fileResult: filesExtensionsFor3PStorageproviders.FileResult,
+    ): void {
+      if (callback) {
+        if (fileResult && fileResult.error) {
+          callback([], fileResult.error);
+        } else {
+          if (fileResult && fileResult.fileChunk) {
+            try {
+              const assemble: filesExtensionsFor3PStorageproviders.AssembleAttachment = decodeAttachment(
+                fileResult.fileChunk,
+                fileResult.fileType,
+              );
+              helper.assembleAttachment.push(assemble);
+
+              if (fileResult.fileChunk.chunkSequence == Number.MAX_SAFE_INTEGER) {
+                const fileBlob = createFile(helper.assembleAttachment, fileResult.fileType);
+
+                if (fileResult.isLastFile) {
+                  // conver blob to File
+                  const receivedFile = new File([fileBlob], fileResult.fileName, {
+                    type: fileBlob.type,
+                  });
+
+                  files.push(receivedFile);
+
+                  callback(files, fileResult.error);
+                }
+
+                helper = {
+                  fileType: '',
+                  assembleAttachment: [],
+                };
+              }
+            } catch (e) {
+              callback([], { errorCode: ErrorCode.INTERNAL_ERROR, message: e });
+            }
+          } else {
+            callback([], { errorCode: ErrorCode.INTERNAL_ERROR, message: 'data received is null' });
+          }
+        }
+      }
+    }
+    sendMessageToParent('getDragAndDropFiles', [dragAndDropInput], handleGetDragAndDropFilesCallbackRequest);
+  }
+
+  /**
+   * @hidden
+   * Helper function to create a blob from file chunks based on their sequence
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export function createFile(
+    assembleAttachment: filesExtensionsFor3PStorageproviders.AssembleAttachment[],
+    mimeType: string,
+  ): Blob {
+    if (assembleAttachment == null || mimeType == null || assembleAttachment.length <= 0) {
+      return null;
+    }
+    let file: Blob | undefined;
+    let sequence = 0;
+    assembleAttachment.sort((a, b) => (a.sequence > b.sequence ? 1 : -1));
+    assembleAttachment.forEach((item) => {
+      if (item.sequence == sequence) {
+        if (file) {
+          file = new Blob([file, item.file], { type: mimeType });
+        } else {
+          file = new Blob([item.file], { type: mimeType });
+        }
+        sequence++;
```

Comment:
Why not use the iterator? Is there a specific reason to update the sequence manually ?

Reply:
Using createFile function from mediaUtils.ts